app/src/main/java/com/eis/phoneringer/MainActivity.java

```diff
@@ -1,14 +1,127 @@
 package com.eis.phoneringer;
 
+import androidx.annotation.NonNull;
+import androidx.annotation.RequiresApi;
 import androidx.appcompat.app.AppCompatActivity;
 
+import android.Manifest;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.os.Build;
 import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
 
+import com.eis.phoneringer.structure.AppManager;
+import com.eis.phoneringer.structure.PasswordManager;
+import com.eis.phoneringer.structure.ReceivedMessageListener;
+import com.eis.phoneringer.structure.RingCommand;
+import com.eis.smslibrary.SMSHandler;
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+import com.eis.smslibrary.exceptions.InvalidSMSMessageException;
+import com.eis.smslibrary.exceptions.InvalidTelephoneNumberException;
+import com.eis.smslibrary.listeners.SMSSentListener;
+
+/**
+ * Test app: through this MainActivity we can test out our basic library for sending ring commands
+ *
+ * @author Alberto Ursino
+ */
 public class MainActivity extends AppCompatActivity {
 
+    private static Button ringButton = null;
+    private static EditText phoneNumber = null;
+    private static final String DEFAULT_PASSWORD = "password";
+    private PasswordManager passwordManager = null;
+
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        final Context context = getApplicationContext();
+        ringButton = findViewById(R.id.ring_button);
+        phoneNumber = findViewById(R.id.phone_number);
+        passwordManager = new PasswordManager(context);
+
+        /**
+         * Some tricks with permissions
+         */
+        if (checkPermission()) {
+            SMSHandler.getInstance().setup(context);
+            SMSHandler.getInstance().setReceivedListener(new ReceivedMessageListener(context));
+        } else {
+            requestPermission();
+        }
+
+        /**
+         * Default password equals for all the devices
+         */
+        if (passwordManager.getPassword().isEmpty())
+            passwordManager.setPassword(DEFAULT_PASSWORD);
+
+        ringButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                sendCommand(context);
+            }
+        });
+    }
+
+    /**
+     * Sends a ring command to the phone number given by the user
+     *
+     * @param context The current application context
+     */
+    private void sendCommand(final Context context) {
```

Comment:
why does this method require passing a Context? since it's part of an Activity it can use the Activity itself as a Context, or get the application context.

Reply:
You are in right but I didn't want to write everytime `getApplicationCode()`

Reply:
Assign the context once and use that variable